non-critical-infra/modules/hydra-queue-runner-v2.nix

```diff
@@ -0,0 +1,260 @@
+            };
+            machineSortFn = lib.mkOption {
+              description = "Function name for sorting machines";
+              type = lib.types.enum [
+                "SpeedFactorOnly"
+                "CpuCoreCountWithSpeedFactor"
+                "BogomipsWithSpeedFactor"
+              ];
+              default = "SpeedFactorOnly";
+            };
+            dispatchTriggerTimerInS = lib.mkOption {
+              description = "Timer for triggering dispatch in an interval in seconds. Setting this to a value <= 0 will disable this timer and only trigger the dispatcher if queue changes happend.";
+              type = lib.types.int;
+              default = 120;
+            };
+            remoteStoreAddr = lib.mkOption {
+              description = "Remote store address";
+              type = lib.types.nullOr lib.types.singleLineStr;
+              default = null;
+            };
+            signingKeyPath = lib.mkOption {
+              description = "Signing key path";
+              type = lib.types.nullOr lib.types.path;
+              default = null;
+            };
+            useSubstitutes = lib.mkOption {
+              description = "Use substitution for paths";
+              type = lib.types.bool;
+              default = false;
+            };
+            rootsDir = lib.mkOption {
+              description = "Gcroots directory, defaults to /nix/var/nix/gcroots/per-user/$LOGNAME/hydra-roots";
+              type = lib.types.nullOr lib.types.path;
+              default = null;
+            };
+            maxRetries = lib.mkOption {
+              description = "Number of maximum amount of retries for a build step.";
+              type = lib.types.ints.positive;
+              default = 5;
+            };
+            retryInterval = lib.mkOption {
+              description = "Interval in which retires should be able to be attempted again.";
+              type = lib.types.ints.positive;
+              default = 60;
+            };
+            retryBackoff = lib.mkOption {
+              description = "Additional backoff on top of the retry interval.";
+              type = lib.types.float;
+              default = 3.0;
+            };
+          };
+        };
+        default = { };
+      };
+
+      grpcAddress = lib.mkOption {
+        type = lib.types.singleLineStr;
+        default = "[::1]";
+        description = "The IP address the grpc listener should bound to";
+      };
+
+      grpcPort = lib.mkOption {
+        description = "Which grpc port this app should listen on";
+        type = lib.types.port;
+        default = 50051;
+      };
+
+      restAddress = lib.mkOption {
+        type = lib.types.singleLineStr;
+        default = "[::1]";
+        description = "The IP address the rest listener should bound to";
+      };
+
+      restPort = lib.mkOption {
+        description = "Which rest port this app should listen on";
+        type = lib.types.port;
+        default = 8080;
+      };
```

Comment:
let's put these into a `grpc` and `rest` attrset, so all aspects are neatly separated like mtls.